### PR TITLE
[Commands] Warn if no test cases match with filter option

### PR DESCRIPTION
**Pull request by bhargavg**

### CODE REVIEW

File: Tests/FunctionalTests/MiscellaneousTests.swift [hunk 1]
```diff
@@ -392,6 +392,15 @@ class MiscellaneousTestCase: XCTestCase {
         #endif
     }
 
+    func testSwiftTestFilterWithNoResults() throws {
```
Comment (aciidb0mb3r) [inline Tests/FunctionalTests/MiscellaneousTests.swift]: I don't think this warning is worth a functional test :/

File: Sources/Commands/SwiftTestTool.swift [hunk 2]
```diff
@@ -140,11 +140,16 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
...
                 }
                 let tests = testSuites.filteredTests(specifier: options.testCaseSpecifier)
-                for test in tests {
-                    let runner = TestRunner(path: testPath,
-                                            xctestArg: test.specifier,
-                                            processSet: processSet)
-                    ranSuccessfully = ranSuccessfully && runner.test()
+
+                if tests.isEmpty {
+                    print("warning: the filter predicate did not match any test case")
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTestTool.swift]: We should us the diagnostic engine

**Comment by hartbit** [comment]
PS: Random stupid question: is the filter still limited to the `<test-target>.<test-case>` or `<test-target>.<test-case>/<test>` as suggested by `--help`? If yes, it might be worth to create a `TestCasePath` type conforming to `ArgumentKind` to validate the format. If not, we might want to update the usage documentation.

**Comment by aciidb0mb3r** [comment]
@hartbit --filter supports regex, so `TestCasePath` doesn't work in this case. It would have been nice to have that when we just had that format.

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**